File: 025.rb
# What is the first term in the Fibonacci sequence to contain 1000 digits?

fibonacci1 = fibonacci2 = term = 1
while true
    break if fibonacci1.to_s.length >= 1000
    fibonacci1, fibonacci2 = fibonacci2, fibonacci2 + fibonacci1
    term += 1
end
puts term

File: clojure/src/project_euler_clojure/025.clj
;; What is the first term in the Fibonacci sequence to contain 1000 digits?

(ns project-euler-clojure.025)

(defn fib [a b]
  (->> (fib b (+' a b))
       (cons a)
       (lazy-seq)))

(defn count-digits [n]
  (->> (str n)
       (count)))

(defn problem-025 []
  (->> (fib 1 1)
       (take-while #(< (count-digits %) 1000))
       (count)
       (inc)))
